test/core/transport/binder/end2end/fuzzers/client_fuzzer.cc

```diff
@@ -12,228 +12,24 @@
-    consumed_data_size_ += data->size();
-    return absl::OkStatus();
-  }
-
- private:
-  FuzzedDataProvider* data_provider_;
-  // Whether this parcel contains a SETUP_TRANSPORT request. If it is, we will
-  // avoid returning errors in the Read* functions so that the fuzzer will not
-  // be blocked waiting for the correct request.
-  bool is_setup_transport_;
-
-  static constexpr size_t kParcelDataSizeLimit = 1024 * 1024;
-  size_t consumed_data_size_;
-};
-
-std::thread* g_fuzzing_thread = nullptr;
-
-void FuzzingLoop(
-    const uint8_t* data, size_t size,
-    grpc_core::RefCountedPtr<grpc_binder::WireReader> wire_reader_ref,
-    grpc_binder::TransactionReceiver::OnTransactCb callback) {
-  FuzzedDataProvider data_provider(data, size);
-  {
-    // Send SETUP_TRANSPORT request.
-    std::unique_ptr<grpc_binder::ReadableParcel> parcel =
-        absl::make_unique<ReadableParcelForFuzzing>(
-            &data_provider,
-            /*is_setup_transport=*/true);
-    callback(static_cast<transaction_code_t>(
-                 grpc_binder::BinderTransportTxCode::SETUP_TRANSPORT),
-             parcel.get())
-        .IgnoreError();
-  }
-  while (data_provider.remaining_bytes() > 0) {
-    transaction_code_t tx_code =
-        data_provider.ConsumeIntegralInRange<transaction_code_t>(
-            0, LAST_CALL_TRANSACTION);
-    std::unique_ptr<grpc_binder::ReadableParcel> parcel =
-        absl::make_unique<ReadableParcelForFuzzing>(
-            &data_provider,
-            /*is_setup_transport=*/false);
-    callback(tx_code, parcel.get()).IgnoreError();
-  }
-  wire_reader_ref = nullptr;
-}
-
-// TransactionReceiver implementation used in fuzzing.
-//
-// When constructed, start sending fuzzed requests to the client. When all the
-// bytes are consumed, the reference to WireReader will be released.
-class TransactionReceiverForFuzzing : public grpc_binder::TransactionReceiver {
- public:
-  TransactionReceiverForFuzzing(
-      const uint8_t* data, size_t size,
-      grpc_core::RefCountedPtr<grpc_binder::WireReader> wire_reader_ref,
-      grpc_binder::TransactionReceiver::OnTransactCb cb) {
-    gpr_log(GPR_INFO, "Construct TransactionReceiverForFuzzing");
-    GPR_ASSERT(g_fuzzing_thread == nullptr);
-    g_fuzzing_thread = new std::thread(
-        FuzzingLoop, data, size, std::move(wire_reader_ref), std::move(cb));
-  }
-
-  void* GetRawBinder() override { return nullptr; }
-};
-
-std::unique_ptr<grpc_binder::TransactionReceiver>
-BinderForFuzzing::ConstructTxReceiver(
-    grpc_core::RefCountedPtr<grpc_binder::WireReader> wire_reader_ref,
-    grpc_binder::TransactionReceiver::OnTransactCb cb) const {
-  return absl::make_unique<TransactionReceiverForFuzzing>(data_, size_,
-                                                          wire_reader_ref, cb);
-}
-
-void* tag(intptr_t t) { return reinterpret_cast<void*>(t); }
-void dont_log(gpr_log_func_args*) {}
+static void* tag(intptr_t t) { return reinterpret_cast<void*>(t); }
```

Comment:
Why not anonymous namespace

Reply:
These are copied from HTTP/2 fuzzers and I feel like it's not necessarily good to change them? It was probably a mistake to put them in anonymous namespace before.

Reply:
Hmm I think our team generally prefers anonymous namespace more but it is ok if HTTP/2 fuzzers are using static

Reply:
Yeah I guess that's because it was written before transitioning to C++.